src/timeseriesinsights/azext_timeseriesinsights/custom.py

```diff
@@ -327,121 +290,31 @@ def timeseriesinsights_event_source_iot_hub_update(client,
                          event_source_update_parameters=patch_parameters)
 
 
-def timeseriesinsights_reference_data_set_list(client,
+def timeseriesinsights_reference_data_set_list(cmd,
                                                resource_group_name,
                                                environment_name):
-    return client.list_by_environment(resource_group_name=resource_group_name,
-                                      environment_name=environment_name).value
-
-
-def timeseriesinsights_reference_data_set_show(client,
-                                               resource_group_name,
-                                               environment_name,
-                                               reference_data_set_name):
-    return client.get(resource_group_name=resource_group_name,
-                      environment_name=environment_name,
-                      reference_data_set_name=reference_data_set_name)
+    List = _ReferenceDataSetList(cmd.loader)
+    parameters = {}
+    parameters['resource_group'] = resource_group_name
+    parameters['environment_name'] = environment_name
+    return List(parameters)["value"]
```

Comment:
Why not use list directly?

Reply:
Need to get .value as the List command does not support pagination.